service/internal/k8sclient/cache.go

```diff
@@ -0,0 +1,143 @@
+package k8sclient
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/kubeconfig/v4"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	gocache "github.com/patrickmn/go-cache"
+	"github.com/spf13/afero"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	expiration = 1 * time.Hour
```

Comment:
I think this TTL is a bit long. The problem is cert renewal for the app-operator cert as I think then the operator won't be able to connect.

What about reducing it to 10 mins or even 5 mins?

Reply:
I think reducing it to 10 mins makes more sense.

Reply:
Cool thanks 👍